controller/webhook/launcher.go

```diff
@@ -0,0 +1,64 @@
+package webhook
+
+import (
+	"flag"
+	"os"
+	"os/signal"
+	"strconv"
+
+	"github.com/linkerd/linkerd2/controller/k8s"
+	"github.com/linkerd/linkerd2/pkg/admin"
+	"github.com/linkerd/linkerd2/pkg/flags"
+	"github.com/linkerd/linkerd2/pkg/tls"
+	log "github.com/sirupsen/logrus"
+)
+
+// Launch sets up and starts the webhook and metrics servers
+func Launch(config *Config) {
+	p := strconv.FormatUint(uint64(config.MetricsPort), 10)
+	metricsAddr := flag.String("metrics-addr", ":"+p, "address to serve scrapable metrics on")
+	addr := flag.String("addr", ":8443", "address to serve on")
+	kubeconfig := flag.String("kubeconfig", "", "path to kubeconfig")
+	controllerNamespace := flag.String("controller-namespace", "linkerd", "namespace in which Linkerd is installed")
+	flags.ConfigureAndParse()
+
+	stop := make(chan os.Signal, 1)
+	defer close(stop)
+	signal.Notify(stop, os.Interrupt, os.Kill)
+
+	k8sAPI, err := k8s.InitializeAPI(*kubeconfig, k8s.NS, k8s.RS)
+	if err != nil {
+		log.Fatalf("failed to initialize Kubernetes API: %s", err)
+	}
+
+	rootCA, err := tls.GenerateRootCAWithDefaults(config.WebhookServiceName)
+	if err != nil {
+		log.Fatalf("failed to create root CA: %s", err)
+	}
+
+	config.api = k8sAPI
+	config.controllerNamespace = *controllerNamespace
+	config.rootCA = rootCA
+
+	selfLink, err := config.Create()
+	if err != nil {
+		log.Fatalf("failed to create the webhook configurations resource: %s", err)
+	}
+	log.Infof("created webhook configuration: %s", selfLink)
+
+	s, err := NewServer(k8sAPI, *addr, config.WebhookServiceName, *controllerNamespace, rootCA, config.Handler)
```

Comment:
if `config.Handler` only exists to get passed through `Launch()` into `NewServer()`, consider just making it a separate param on `Launch()`. similar question for `MetricsPort` and others.

Reply:
I kinda did that to avoid having to pass so many arguments, but I think I can reach a compromise in my next push :+1: